apps/nsqd/nsqd.go

```diff
@@ -88,6 +88,8 @@ func nsqFlagset() *flag.FlagSet {
 
 	authHTTPAddresses := app.StringArray{}
 	flagSet.Var(&authHTTPAddresses, "auth-http-address", "<addr>:<port> to query auth server (may be given multiple times)")
+	flagSet.Bool("http-auth-required", false, "require auth for HTTP clients")
+	flagSet.Bool("https-auth-required", false, "require auth for HTTPS clients")
```

Comment:
That's kinda where I was headed too, let's experiment with that approach?

FWIW, it's consistent with how `--e2e-processing-latency-percentile` works.